src/dmd/semantic3.d

```diff
@@ -1146,14 +1146,16 @@ private extern(C++) final class Semantic3Visitor : Visitor
 
                             s = s.statementSemantic(sc2);
 
-                            bool isnothrow = f.isnothrow & !(funcdecl.flags & FUNCFLAG.nothrowInprocess);
+                            immutable bool isnothrow = f.isnothrow && !(funcdecl.flags & FUNCFLAG.nothrowInprocess);
                             const blockexit = s.blockExit(funcdecl, isnothrow);
                             if (blockexit & BE.throw_)
+                            {
                                 funcdecl.eh_none = false;
-                            if (f.isnothrow && isnothrow && blockexit & BE.throw_)
-                                error(funcdecl.loc, "`nothrow` %s `%s` may throw", funcdecl.kind(), funcdecl.toPrettyChars());
-                            if (funcdecl.flags & FUNCFLAG.nothrowInprocess && blockexit & BE.throw_)
-                                f.isnothrow = false;
+                                if (isnothrow)
+                                    error(funcdecl.loc, "`nothrow` %s `%s` may throw", funcdecl.kind(), funcdecl.toPrettyChars());
+                                else if(funcdecl.flags & FUNCFLAG.nothrowInprocess)
```

Comment:
space between `if(fun`

Reply:
Also the `else` is likely to change the behavior in cascading errors

Reply:
`isnothrow` check for negated `funcdecl.flags & FUNCFLAG.nothrowInprocess`, so if `isnothrow` is true, the next if is always false, therefore the `else` here is logically correct.

Reply:
> space between `if(fun`

Changed.